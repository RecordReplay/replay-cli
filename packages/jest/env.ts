import ReplayEnvironment from "./src/env";

export default ReplayEnvironment;
