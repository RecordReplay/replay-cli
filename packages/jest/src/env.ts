import { Event, State } from "jest-circus";
import NodeEnvironment from "jest-environment-node";

class ReplayEnvironment extends NodeEnvironment {
  testPath?: string;

  constructor(config: any, context: any) {
    super(config);
    this.testPath = context.testPath;
    console.log(">>>>>>", this.testPath);
  }

  async handleTestEvent(event: Event, state: State) {
    if (event.name === "test_start") {
      process.env;
    }
  }
}

export default ReplayEnvironment;
