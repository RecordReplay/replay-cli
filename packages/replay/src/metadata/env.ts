export * from "@replay-cli/shared/recording/metadata/legacy/env";
