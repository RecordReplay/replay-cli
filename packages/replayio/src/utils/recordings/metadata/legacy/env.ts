import { defaulted, string } from "superstruct";

type Resolver = string | ((env: NodeJS.ProcessEnv) => string | undefined);

const firstEnvValueOf =
  (...envKeys: Resolver[]) =>
  () =>
    envKeys.reduce<string | undefined>(
      (a, k) => a || (typeof k === "function" ? k(process.env) : process.env[k]),
      undefined
    );

const envString = (...envKeys: Resolver[]) => defaulted(string(), firstEnvValueOf(...envKeys));

export { firstEnvValueOf, envString };
