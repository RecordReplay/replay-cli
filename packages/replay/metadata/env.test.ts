import { firstEnvValueOf } from "./env";

describe("firstEnvValueOf", () => {
  let env: NodeJS.ProcessEnv;

  beforeEach(() => {
    env = process.env;
    process.env = {};
  });

  afterEach(() => {
    process.env = env;
  });

  describe("using key names", () => {
    it("returns first value", () => {
      process.env = {
        KEY_1: "key 1",
        KEY_2: "key 2",
      };

      const resolver = firstEnvValueOf("KEY_1", "KEY_2");
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("ignores empty strings", () => {
      process.env = {
        KEY_1: "",
        KEY_2: "key 2",
      };

      const resolver = firstEnvValueOf("KEY_1", "KEY_2");
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("ignores undefined keys", () => {
      process.env = {
        KEY_1: undefined,
        KEY_2: "key 2",
      };

      const resolver = firstEnvValueOf("KEY_1", "KEY_2");
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("returns undefined when no keys match", () => {
      process.env = {
        KEY_1: undefined,
        KEY_2: "key 2",
      };

      const resolver = firstEnvValueOf("KEY_3");
      const result = resolver();

      expect(result).toMatchSnapshot();
    });
  });

  describe("using callbacks", () => {
    it("returns first value", () => {
      process.env = {
        KEY_1: "key 1",
        KEY_1_VALUE: "value 1",
        KEY_2: "key 2",
        KEY_2_VALUE: "value 2",
      };

      const resolver = firstEnvValueOf(
        env => env.KEY_1 && env.KEY_1_VALUE,
        env => env.KEY_2 && env.KEY_2_VALUE
      );
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("ignores empty strings", () => {
      process.env = {
        KEY_1: "key 1",
        KEY_1_VALUE: "",
        KEY_2: "key 2",
        KEY_2_VALUE: "value 2",
      };

      const resolver = firstEnvValueOf(
        env => env.KEY_1 && env.KEY_1_VALUE,
        env => env.KEY_2 && env.KEY_2_VALUE
      );
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("ignores undefined keys", () => {
      process.env = {
        KEY_1: "key 1",
        KEY_1_VALUE: undefined,
        KEY_2: "key 2",
        KEY_2_VALUE: "value 2",
      };

      const resolver = firstEnvValueOf(
        env => env.KEY_1 && env.KEY_1_VALUE,
        env => env.KEY_2 && env.KEY_2_VALUE
      );
      const result = resolver();

      expect(result).toMatchSnapshot();
    });

    it("returns undefined when no keys match", () => {
      process.env = {
        KEY_1: "key 1",
        KEY_2: "key 2",
      };

      const resolver = firstEnvValueOf(
        env => env.KEY_1 && env.KEY_1_VALUE,
        env => env.KEY_2 && env.KEY_2_VALUE
      );
      const result = resolver();

      expect(result).toMatchSnapshot();
    });
  });
});
