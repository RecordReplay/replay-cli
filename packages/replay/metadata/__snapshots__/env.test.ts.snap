// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`firstEnvValueOf using callbacks ignores empty strings 1`] = `"value 2"`;

exports[`firstEnvValueOf using callbacks ignores undefined keys 1`] = `"value 2"`;

exports[`firstEnvValueOf using callbacks returns first value 1`] = `"value 1"`;

exports[`firstEnvValueOf using callbacks returns undefined when no keys match 1`] = `undefined`;

exports[`firstEnvValueOf using key names ignores empty strings 1`] = `"key 2"`;

exports[`firstEnvValueOf using key names ignores undefined keys 1`] = `"key 2"`;

exports[`firstEnvValueOf using key names returns first value 1`] = `"key 1"`;

exports[`firstEnvValueOf using key names returns undefined when no keys match 1`] = `undefined`;
